software/controller/lib/hal/stepper.cpp

```diff
@@ -218,43 +219,13 @@ void StepMotor::OneTimeInit() {
 
   // DMA2 channels 3 and 4 can be used to handle rx and tx interrupts from
   // SPI1 respectively.
-  DMA::SelectChannel(DMA::Base::DMA2, DMA::Channel::Chan3, 4);
-  DMA::SelectChannel(DMA::Base::DMA2, DMA::Channel::Chan4, 4);
-
-  // The two DMA channels move data to/from the SPI data register.
-  DmaReg *dma = DMA::get_register(DMA::Base::DMA2);
-
-  // Configure the DMA channels, but don't enable them yet
-  /// \TODO: (subtly) repetitive blocks for each channel - factor this out
-  int c3 = static_cast<int>(DMA::Channel::Chan3);
-  dma->channel[c3].peripheral_address = &spi->data;
-  dma->channel[c3].config.enable = 0;
-  dma->channel[c3].config.tx_complete_interrupt = 1;
-  dma->channel[c3].config.half_tx_interrupt = 0;
-  dma->channel[c3].config.tx_error_interrupt = 0;
-  dma->channel[c3].config.direction = static_cast<uint32_t>(DMA::ChannelDir::PeripheralToMemory);
-  dma->channel[c3].config.circular = 0;
-  dma->channel[c3].config.peripheral_increment = 0;
-  dma->channel[c3].config.memory_increment = 1;
-  dma->channel[c3].config.peripheral_size = 0;
-  dma->channel[c3].config.memory_size = 0;
-  dma->channel[c3].config.priority = 0;
-  dma->channel[c3].config.mem2mem = 0;
-
-  int c4 = static_cast<int>(DMA::Channel::Chan4);
-  dma->channel[c4].peripheral_address = &spi->data;
-  dma->channel[c4].config.enable = 0;
-  dma->channel[c4].config.tx_complete_interrupt = 0;
-  dma->channel[c4].config.half_tx_interrupt = 0;
-  dma->channel[c4].config.tx_error_interrupt = 0;
-  dma->channel[c4].config.direction = static_cast<uint32_t>(DMA::ChannelDir::MemoryToPeripheral);
-  dma->channel[c4].config.circular = 0;
-  dma->channel[c4].config.peripheral_increment = 0;
-  dma->channel[c4].config.memory_increment = 1;
-  dma->channel[c4].config.peripheral_size = 0;
-  dma->channel[c4].config.memory_size = 0;
-  dma->channel[c4].config.priority = 0;
-  dma->channel[c4].config.mem2mem = 0;
+  rx_dma_.emplace(DMA::Base::DMA2, DMA::Channel::Chan3);
+  tx_dma_.emplace(DMA::Base::DMA2, DMA::Channel::Chan4);
+
+  rx_dma_->Initialize(4, &spi->data, DMA::ChannelDir::PeripheralToMemory, /*tx_interrupt=*/true,
```

Comment:
Add comment for parameter name of `4`. Or constant, like in HAL comment above.

Reply:
Same here.